### PR TITLE
arch/x86: Generalize cache manipulation exposure

### PR DESCRIPTION
**Pull request by tbursztyka**
Generalize cache exposure from what used to be in ia32 only before.

### CODE REVIEW

File: arch/x86/core/intel64/cache_s.S [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+.global z_is_clflush_available
+z_is_clflush_available:
+	pushq %rbx
+	movq $1, %rax
+	cpuid
+	movq %rdx, %rax
+	andq $CPUID_CFLSH_BIT, %rax
+	popq %rbx
+	ret
```
Comment (andyross) [inline arch/x86/core/intel64/cache_s.S:31]: Mild complaint: runtime feature detection isn't really a Zephyr thing.  Obviously x86 is right at the boundary where sometimes you have to, but in this case?  I dunno, cache line size is 512 bytes across the whole universe now and CLFLUSH works everywhere too.  I honestly think you could skip this, and if we really need it across architectures we have in-tree it belongs in devicetree instead (and I say that as someone who really hates devicetree and loves x86 assembly!)

**Review by andyross — CHANGES_REQUESTED** [review]
Agree with the API motion, mildly opposed to detection code.  But I'm willing to be overruled.

File: arch/x86/core/intel64/cache_s.S (hunk 1 above)
Comment (tbursztyka) [reply arch/x86/core/intel64/cache_s.S:31]: Ok that means updating ia32 as well then (I just c/p cache_s.S from there with minor modifications)

**Review by andyross — APPROVED** [review]
Sorry, I didn't see your reply until you had already made the change.  I didn't realize this was a pre-existing API.   Still, I like this version better anyway (but would be fine with original too)

**Review by jhedberg — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**